lib/cylc/cycling/iso8601.py

```diff
@@ -288,9 +293,12 @@ class ISO8601Exclusions(ExclusionBase):
     grouped exclusion sequences. The Python ``in`` and ``not in`` operators
     may be used on this object to determine if a point is in the collection
     of exclusion sequences."""
+
+    __slots__ = ExclusionBase.__slots__ + ('p_iso_exclusions',)
+
     def __init__(self, excl_points, start_point, end_point=None):
         super(ISO8601Exclusions, self).__init__(start_point, end_point)
-        self.p_iso_exclusions = set()
+        self.p_iso_exclusions = []
```

Comment:
The reason I changed this is because an empty set is much larger than an empty list:

Size in bytes for collections of `None` elements:

elements | list | set
--- | --- | ---
0 | 72 | 232
1 | 96 | 248
2 | 104 | 248
3 | 112 | 248
4 | 120 | 248
5 | 128 | 248
6 | 136 | 248
7 | 144 | 248
8 | 152 | 248
9 | 160 | 248